# Introduction

This analysis will look at the Euro/USD exchange rate and find evidence to correlate how it can be affected by world events that stem sometimes from here at home in the United States or other nations. There will be instances where the events might make you think that the exchange should be dramatic but the data will say otherwise! The data being used primarily is the exchange rate of the Euro vs. 1 US Dollar between the dates of 11/30/2003 and 7/29/2022. The other data set with important world events is a csv I created by mapping out dates for important US and World Events with the region where the origination of those very events occur.

This Jupyter notebook takes the csv files, cleans and merges them, then exports them as csv files in the Results folder to be used for visualization in Tableau, which you can access through a link in the readme file.

### Import Pandas

In [1]:
import pandas as pd
import warnings
warnings.filterwarnings("ignore")

### Load first set of data

In [2]:
# Reading my first csv file, the Euro/USD exchange rates
euro_df = pd.read_csv('Input/EURUSDX.csv', usecols = ['Date', 'Close'])

euro_df.head()

Date     Close
0  2003-12-01  1.196501
1  2003-12-02  1.208897
2  2003-12-03  1.212298
3  2003-12-04  1.208094
4  2003-12-05  1.218695

In [3]:
# Checking basic info such as shape, NaNs and dtypes
euro_df.shape

(4870, 2)

In [4]:
# Checking for null values in the euro dataframe
euro_df.isnull().values.any()

True

In [5]:
# Pulling information about the data frame such as number of columns, column labels, etc.
euro_df.info

<bound method DataFrame.info of             Date     Close
0     2003-12-01  1.196501
1     2003-12-02  1.208897
2     2003-12-03  1.212298
3     2003-12-04  1.208094
4     2003-12-05  1.218695
...          ...       ...
4865  2022-07-25  1.020023
4866  2022-07-26  1.022495
4867  2022-07-27  1.012956
4868  2022-07-28  1.020929
4869  2022-07-29  1.024380

[4870 rows x 2 columns]>

In [6]:
# Calculating some statistical data about the euro dataframe
euro_df.describe()

Close
count  4841.000000
mean      1.250149
std       0.121111
min       1.002567
25%       1.140004
50%       1.238497
75%       1.336791
max       1.598798

### Load second set of data

In [7]:
# Reading my second csv file, the important world event dates
dates_df = pd.read_csv('Input/World_Events.csv', usecols = ['Event', 'Date', 'Region'])

dates_df.head()


Event        Date Region
0                           Iraq War  2003-03-20     US
1         2004 Presidential Election  2004-11-02     US
2  Indian Ocean Earthquake & Tsunami  2004-12-26  WORLD
3            Pope John Paul II Death  2005-04-02  WORLD
4                iPhone Introduction  2007-01-09  WORLD

### Test run

In [8]:
# Test to see if a random date gives me the correct outcome
euro_2015 = euro_df.loc[euro_df.Date == '2015-01-01']

print(euro_2015)


            Date     Close
2893  2015-01-01  1.209863


In [9]:
# Another test to see if the dates dataframe gives me the correct outcome for a random date
event_1 = dates_df.loc[dates_df.Date == '2004-11-02']

print(event_1)

                        Event        Date Region
1  2004 Presidential Election  2004-11-02     US


### Clean and Merge data

In [10]:
# Merged dataframes to give an overview of various events and the closing rate of the Euro/USD exchange
euro_events = euro_df.merge(dates_df)

print(euro_events)

          Date     Close                                 Event Region
0   2004-11-02  1.272006            2004 Presidential Election     US
1   2007-01-09  1.299697                   iPhone Introduction  WORLD
2   2008-09-15  1.426900                  2008 Great Recession  WORLD
3   2008-09-29  1.439325                  Housing Market Crash     US
4   2008-11-04  1.298398            2008 Presidential Election     US
5   2008-12-11  1.332996                        Madoff Scandal     US
6   2010-12-17  1.323995                           Arab Spring  WORLD
7   2011-01-14  1.335327                      Tunisia Uprising  WORLD
8   2011-02-03  1.381788                        Yemen Uprising  WORLD
9   2011-02-09  1.362695                        Egypt Uprising  WORLD
10  2011-02-15  1.348109                        Libya Uprising  WORLD
11  2011-03-11  1.380129    2011 Japanese Earthquake & Tsunami  WORLD
12  2011-04-29  1.484406                        Syria Uprising  WORLD
13  2012-05-18  1.26

### Filtering data

In [11]:
# Results for Euro/USD exchange rate based on World only events
world_events = euro_events.loc[euro_events.Region == 'WORLD']

world_events.head()

Date     Close                 Event Region
1  2007-01-09  1.299697   iPhone Introduction  WORLD
2  2008-09-15  1.426900  2008 Great Recession  WORLD
6  2010-12-17  1.323995           Arab Spring  WORLD
7  2011-01-14  1.335327      Tunisia Uprising  WORLD
8  2011-02-03  1.381788        Yemen Uprising  WORLD

In [12]:
# Results for Euro/USD exchange rate based on US only events
us_events = euro_events.loc[euro_events.Region == 'US']

us_events.head()

Date     Close                       Event Region
0   2004-11-02  1.272006  2004 Presidential Election     US
3   2008-09-29  1.439325        Housing Market Crash     US
4   2008-11-04  1.298398  2008 Presidential Election     US
5   2008-12-11  1.332996              Madoff Scandal     US
13  2012-05-18  1.269841                Facebook IPO     US

### Cleaning and calculating new values

In [13]:
# Cleaning data to see results for how Brexit affected the Euro/USD exchange rate
brexit_status = ['Brexit Talks Start', 'Brexit Implemented']
mask = euro_events['Event'].isin(brexit_status)
brexit_events = euro_events[mask]

print(brexit_events)


          Date     Close               Event Region
17  2017-03-29  1.081771  Brexit Talks Start  WORLD
19  2020-01-31  1.102913  Brexit Implemented  WORLD


In [14]:
euro_df['3-Day Rolling Average'] = euro_df['Close'].rolling(window=3).mean()

print(euro_df)

            Date     Close  3-Day Rolling Average
0     2003-12-01  1.196501                    NaN
1     2003-12-02  1.208897                    NaN
2     2003-12-03  1.212298               1.205899
3     2003-12-04  1.208094               1.209763
4     2003-12-05  1.218695               1.213029
...          ...       ...                    ...
4865  2022-07-25  1.020023               1.020136
4866  2022-07-26  1.022495               1.021542
4867  2022-07-27  1.012956               1.018491
4868  2022-07-28  1.020929               1.018793
4869  2022-07-29  1.024380               1.019422

[4870 rows x 3 columns]


In [15]:
euro_df['Previous Day Difference'] = euro_df['Close'].diff()

print(euro_df[['Date', 'Close', 'Previous Day Difference']].head(10))

         Date     Close  Previous Day Difference
0  2003-12-01  1.196501                      NaN
1  2003-12-02  1.208897                 0.012396
2  2003-12-03  1.212298                 0.003401
3  2003-12-04  1.208094                -0.004204
4  2003-12-05  1.218695                 0.010601
5  2003-12-08  1.222001                 0.003306
6  2003-12-09  1.224995                 0.002994
7  2003-12-10  1.219096                -0.005899
8  2003-12-11  1.222404                 0.003308
9  2003-12-12  1.227898                 0.005494


In [16]:
euro_df['Percentage Change'] = (euro_df['Close'].pct_change() * 100).round(2).astype(str) + '%'

print(euro_df[['Date', 'Close', 'Percentage Change']].head(10))

         Date     Close Percentage Change
0  2003-12-01  1.196501              nan%
1  2003-12-02  1.208897             1.04%
2  2003-12-03  1.212298             0.28%
3  2003-12-04  1.208094            -0.35%
4  2003-12-05  1.218695             0.88%
5  2003-12-08  1.222001             0.27%
6  2003-12-09  1.224995             0.25%
7  2003-12-10  1.219096            -0.48%
8  2003-12-11  1.222404             0.27%
9  2003-12-12  1.227898             0.45%


### Write to csv

In [17]:
# Exporting results to a csv file for all of the events
euro_events.to_csv('Results/euro_events.csv', index=False)

In [18]:
# Exporting results to a csv file for all world only events
world_events.to_csv('Results/world_only.csv', index=False)

In [19]:
# Exporting results to a csv file for US only events
us_events.to_csv('Results/us_only.csv', index=False)

## Summary of Findings
The Euro when compared to the US Dollar looks to be a steady contender until the 2008s Great Recession and the subsequent US Housing Market Crash a couple weeks later where it nosedived almost 11% from €1.2297 to €1.4393 from the beginning of that year to September of that year. It did, however, recover 2 months later on 11/04/2008 on the day of the US Presidential Election, which saw Barack Hussein Obama as the winner and President Elect. 2 of these 3 important events mentioned have origins primarily from the United States with the Recession also mostly originating from the failed banks in America and creating a domino effect with failed banks around the world that snowballed into the Great Recession. These events point to the fact that the US based events do definitely affect the value and the buying power of the Euro currency.

Following the previous events, the Euro stabilizes in the €1.30~ range for the next few years. It next saw a dramatic drop during the Arab Spring's ongoing uprising, where many Middle Eastern nations had revolutions and government instability for most of 2011 and the Euro peaked at €1.4844, which is it's highest ever when looking at the data within this study, during the Syrian Uprising. This gives us insight on another external region affecting the Euro currency's value and how those forces can change the outlook of a world renowned medium of exchange.

The next important point seen in the cleaned data is on the date 5/18/2012 when Facebook Inc.'s IPO was offered. From here on out, the Euro climbed in value from €1.2802 to €1.0818 when Brexit talks began on 3/29/2017. Brexit was the withdrawal of the United Kingdom from the European Union since being a member state of the European Union's previous predecessor, the European Communities since 1973. This event was very much an internal affair for the European Union and therefore the Euro currency but as the data shows, the valuation of the Euro was not affected much from the date the talks began for Brexit until it was finally implemented almost 3 years later on 1/31/2020. The change in value when compared to an exchange of 1 US Dollar shows a drop of €0.0212 in value during this time.

The 2020 US Presidential Election saw the Euro drop to €1.1642 hinting at the growing instability of US politics. The US Capitol Attack devalued the Euro against the US Dollar almost 6% to €1.2300. President Biden's Inauguration saw the Euro gain 1.5% back in value to €1.2138. 

By the time of the Russian invasion of Ukraine occurred on 2/24/2022, the Euro had recovered from its previous low and gained 7% in value to €1.1305. Following this, 2022s Inflation Rise was reported at the end of the first half of 2022. The inflation rate was reported to be 9.1%, a record high since the early 1980s in the US. Not being influenced by further economic instability in the United States, the Euro saw a gain of 8% in value to €1.0446 during this very time when the highest ever inflation rate in recent history was reported in the United States.

To summarize all of my findings in this study, initially the Euro had been fluctuating in the €1.30 range starting in November of 2004 excluding the dips and gains mentioned earlier but following the 2016 Presidential Election, the Euro saw a dramatic gain in value and has since come very close to being on par with the US Dollar. This study shows that although the US economy, politics and overall instability do greatly affect the worth of the Euro, sometimes even more so than it's own internal challenges, it has seen a steady gain in value and has come out to be a very close contender to the US Dollar on the world stage with its prowess in buying power.